src/ucp/rndv/rndv.c

```diff
@@ -1674,7 +1674,9 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
         ucp_tag_offload_cancel_rndv(sreq);
     }
 
-    ucp_worker_del_request_id(sreq->send.ep->worker, sreq->send.msg_proto.sreq_id);
+    if (ucp_ep_use_indirect_id(ep)) {
+        ucp_worker_del_request_id(sreq->send.ep->worker, sreq->send.msg_proto.sreq_id);
+    }
```

Comment:
according to assert in UCS, `sreq_id == NULL` here? if yes, is it expected by protocol?

Reply:
when tag_offload is used, the msg_proto struct should not be access here, but rather the tag_offload struct (which doesn't have the sreq_id field).
but since tag_offload doesn't support error handling, the del_req_id function "won't do anything" anyway so it should be ok not to call it in this case.

Reply:
so it should be in 
```
if (sreq->flags & UCP_REQUEST_FLAG_OFFLOADED) {
...
} else {
    ucp_worker_del_request_id(sreq->send.ep->worker, sreq->send.msg_proto.sreq_id);
}
```
?

Reply:
checking ucp_ep_use_indirect_id(ep) makes the del_req_id call dependant on err_handling support rather than the used protocol (when checking the sreq->flags)

Reply:
but union in `sreq->send` (to access `sreq->send.msg_proto.sreq_id`) is protocol specific but not err_handleng mode

Reply:
yes but, adding and removing from the ptr_map is err_handling dependant.

Reply:
`ucp_worker_del_request_id` already should handle err handling inside by indirect flag presence, if `sreq_id` is not valid by some other logic - it should be handled accordingly

Reply:
i'm fine with both ways.
@yosefe wdyt?

Reply:
for now let's check for NULL in ptr_map, it's usefult for TCP ep's as well and document it
if we'll have use case where map should hold NULL can move the check to upper layers

Reply:
what do you mean? we can't access msg_proto here for tag_offload